### ISSUE TITLE
Applying gradle-retrolambda spoils classpath when compiling tests - classes from src/main/java are not visible

### ISSUE THREAD

**Issue by mgrzechocinski** [evant/gradle-retrolambda#88]
Hi,

Here is a [sample project](https://github.com/mgrzechocinski/Materializer/tree/topic/unit_test_without_retrolambda) which uses `classpath 'com.android.tools.build:gradle:1.1.0'` with unit tests support.

I wrote a very simple test just to play with it: 
```java
public class MainActivityTest {
     @Test
     public void shouldMainActivityBeNotNull() throws Exception {
           //dummy test
           MainActivity mainActivity = new MainActivity();
           Assertions.assertThat(mainActivity).isNotNull();
     }
}
```
`MainActivityTest` is in the same package as `MainActivity`, but in `src/test/java`. Project does not use gradle-retrolambda plugin. Running `./gradlew clean test` finishes successfuly, test passes. 

```
...
:app:test

BUILD SUCCESSFUL
```
When [I add gradle-retrolambda plugin](https://github.com/mgrzechocinski/Materializer/pull/1/files) my project fails on classpath issues. **Please note that I don't use any Java8 features in both production and test Java code. I only add a plugin to the build process**

```
:app:compileFlavor1DebugUnitTestJava
/Users/mateuszgrzechocinski/dev/RD/Materializer/app/src/test/java/pl/mobilewarsaw/android/materializer/MainActivityTest.java:11: error: cannot find symbol
        MainActivity mainActivity = new MainActivity();
        ^
  symbol:   class MainActivity
  location: class MainActivityTest
/Users/mateuszgrzechocinski/dev/RD/Materializer/app/src/test/java/pl/mobilewarsaw/android/materializer/MainActivityTest.java:11: error: cannot find symbol
        MainActivity mainActivity = new MainActivity();
                                        ^
  symbol:   class MainActivity
  location: class MainActivityTest
2 errors
:app:compileFlavor1DebugUnitTestJava FAILED
```
I guess the problem is that gradle-retrolambda modifies origin `compile<Variant>Java` task, and since `compile<Variant>UnitTestJava` depends on it's output it might not see `*.class` files generated by it. Am I right?

Maybe, once #86 is fixed and gradle-retrolambda plugin modifies origin `compile<Variant>UnitTestJava` task, this problem would dissapear eventually.

Any hints for a workaround for the time being?

**Comment by JoanZapata** [comment]
Okay so I struggled a little bit myself on this and found a temporary workaround.
I made sure RetroLambda is executed **before** unit tests are compiled, using these few lines:

```
gradle.projectsEvaluated {
    def compileTasks = tasks.findAll { it.name.matches("compile.*UnitTestJava") }
    compileTasks.each {
        def match = it.name =~ /compile(.*)UnitTestJava/
        def associatedRetrolambdaCompile = tasks["compileRetrolambda" + match[0][1]];
        it.dependsOn associatedRetrolambdaCompile
    }
}
```

Hope that helps.

**Comment by evant** [comment]
This will be fixed with #86.

**Comment by evant** [comment]
Fixed in 3.0.0
